ci-operator/step-registry/redhat-appstudio/rhtap-installer/sprayproxy-cleanup/redhat-appstudio-rhtap-installer-sprayproxy-cleanup-commands.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export SPRAYPROXY_SERVER_URL \
+  SPRAYPROXY_SERVER_TOKEN
+
+SPRAYPROXY_SERVER_URL=$(cat /usr/local/rhtap-ci-secrets/rhtap/sprayproxy-server-url)
+SPRAYPROXY_SERVER_TOKEN=$(cat /usr/local/rhtap-ci-secrets/rhtap/sprayproxy-server-token)
+
+unregister_pac_server(){
+  webhook_url="$1"
+  echo "Unregistering PAC server [$webhook_url] from SprayProxy server"
+  for _ in {1..5}; do
+    if curl -k -X DELETE -H "Authorization: Bearer ${SPRAYPROXY_SERVER_TOKEN}" "${SPRAYPROXY_SERVER_URL}"/backends --data '{"url": "'"$webhook_url"'"}'; then
+      break
+    fi
+    sleep 5
+  done
+}
+pac_servers=""
+get_pac_servers(){
+  echo "Get PAC servers registered in SprayProxy server"
+  for _ in {1..5}; do
+    if pac_servers=$(curl -k -X GET -H "Authorization: Bearer ${SPRAYPROXY_SERVER_TOKEN}" "${SPRAYPROXY_SERVER_URL}"/backends); then
+      break
+    fi
+    sleep 5
+  done
+}
+
+
+get_pac_servers
+if [[ "$pac_servers" == "Backend urls: " ]]; then
+  echo "No PAC servers registered in SprayProxy server"
+  exit 0
+fi
+
+urls=$(echo "$pac_servers" | grep -o 'https://[^,]*' | sed 's/,//g')
+for url in $urls
+do
```

Comment:
im afraid if we have a cluster running at the same time when we delete the server. Can we remove only latest 24h servers? or at least make a ping to the backend to see is a alive. And in case is not alive dont delete it. WDYT?

Reply:
fixed.